test/test.py

```diff
@@ -85,6 +88,20 @@ def test_fmpz():
                     assert ltype(s) + rtype(t) == s + t
                     assert ltype(s) - rtype(t) == s - t
                     assert ltype(s) * rtype(t) == s * t
+                    assert ltype(s) & rtype(t) == s & t
+                    assert ~ltype(s) == ~s
+
+                    # XXX: Some sort of bug here means that the following fails
+                    # for values of s and t bigger than the word size.
+                    if abs(s) < 2**62 and abs(t) < 2**62:
+                        assert ltype(s) | rtype(t) == s | t
+                    else:
+                        # This still works so somehow the internal representation
+                        # of the fmpz made by fmpz_or is such that fmpz_equal
+                        # returns false even though the values look the same.
+                        assert str(ltype(s) | rtype(t)) == str(s | t)
+
+                    assert ltype(s) ^ rtype(t) == s ^ t
```

Comment:
I'm not sure what is happening here but I think that the `__or__` method here is correct so maybe this is a Flint bug. The `__and__` and `__xor__` methods have the same code and tests and they work fine.

It seems that somehow `fmpz_or` creates an `fmpz` that `fmpz_equal` considers different even though its string representation is the same:
```python
>>> flint.fmpz(2**62) | -1
-1
>>> flint.fmpz(2**62) | -1 == -1
False
>>> flint.fmpz(2**62-1) | -1
-1
>>> flint.fmpz(2**62-1) | -1 == -1
True
```
I'm not sure how to debug this via python-flint.

Reply:
Yeah, this is a bug in ``fmpz_or``. Fixed now in https://github.com/flintlib/flint2/commit/9809014e43c36660e1c8c6484030452928cb534b.

Reply:
Speedy work!